### PR TITLE
Updated replacement logic in release scripts

### PR DESCRIPTION
**Pull request by LorcanMcVeigh**
### Proposed changes
Update release helper scripts:
Update the changelog template - added ubi image.
Fix version replacement logic. 
updated docs-web/technical-specifications.md

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [ ] I have rebased my branch onto master
- [ ] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: hack/changelog-template.txt [hunk 1]
```diff
@@ -9,6 +9,6 @@ HELM CHART:
 * The version of the Helm chart is now %%HELM_CHART_VERSION%%.
 
 UPGRADE:
-* For NGINX, use the %%IC_VERSION%% image from our DockerHub: `nginx/nginx-ingress:%%IC_VERSION%%` or `nginx/nginx-ingress:%%IC_VERSION%%-alpine`
+* For NGINX, use the %%IC_VERSION%% image from our DockerHub: `nginx/nginx-ingress:%%IC_VERSION%%` or `nginx/nginx-ingress:%%IC_VERSION%%-alpine` or 'nginx/nginx-ingress:%%IC_VERSION%%-ubi
```
Comment (pleshakov) [inline hack/changelog-template.txt:12]: too many or. please see the changelog for 1.7.0 -- https://github.com/nginxinc/kubernetes-ingress/releases/tag/v1.7.0

File: hack/prepare-minor-release.sh [hunk 2]
```diff
@@ -36,7 +36,7 @@ fi
 ic_version=$1
 helm_chart_version=$2
 
-prev_ic_version=$(echo $ic_version | awk -F. '{ printf("%s.%s.%d", $1, $2, $3-1) }')
+prev_ic_version=$(echo $ic_version | awk -F. '{ printf("%s\\.%s\\.%d", $1, $2, $3-1) }')
 prev_helm_chart_version=$(echo $helm_chart_version | awk -F. '{ printf("%s.%s.%d", $1, $2, $3-1) }')
```
Comment (pleshakov) [inline hack/prepare-minor-release.sh:40]: is it possible that the same problem can appear for helm chart version replacement as well?

File: hack/prepare-minor-release.sh [hunk 3]
```diff
@@ -51,7 +51,7 @@ sed -i "" "1r hack/changelog-template.txt" $DOCS_TO_UPDATE_FOLDER/releases.md
 sed -i "" -e "s/%%TITLE%%/## NGINX Ingress Controller $ic_version/g" -e "s/%%IC_VERSION%%/$ic_version/g" -e "s/%%HELM_CHART_VERSION%%/$helm_chart_version/g" $DOCS_TO_UPDATE_FOLDER/releases.md
 
 # update docs
-find $DOCS_TO_UPDATE_FOLDER -type f -name "*.md" -exec sed -i "" "s/v$prev_ic_version/v$ic_version/g" {} +
+find $DOCS_TO_UPDATE_FOLDER -type f -name "*.md" -exec sed -i "" "s/$prev_ic_version/$ic_version/g" {} +
```
Comment (pleshakov) [inline hack/prepare-minor-release.sh:54]: this will replace the values in the `docs-web/releases` for previous releases. A bug.

**Comment by pleshakov** [comment]
@LorcanMcVeigh 
thanks the changes. Please make sure the tech-spec page doc is updated. The page mentions images like `nginx/nginx-ingress:1.7.0`

**Comment by LorcanMcVeigh** [comment]
> @LorcanMcVeigh
> thanks the changes. Please make sure the tech-spec page doc is updated. The page mentions images like `nginx/nginx-ingress:1.7.0`

is that not the latest image on dockerhub?

**Pull request merged**